commons/src/main/java/org/eclipse/kapua/commons/service/internal/ServiceDAO.java

```diff
@@ -268,14 +271,21 @@ protected ServiceDAO() {
     /**
      * Find by fields {@link KapuaEntity} utility method
      *
-     * @param em    The {@link EntityManager} that holds the transaction.
-     * @param clazz The {@link KapuaEntity} class. This must be the implementing {@code class}.
-     * @param name  The {@link KapuaEntity} name of the field from which to search.
-     * @param value The value of the field from which to search.
+     * @param em      The {@link EntityManager} that holds the transaction.
+     * @param clazz   The {@link KapuaEntity} class. This must be the implementing {@code class}.
+     * @param scopeId
+     * @param name    The {@link KapuaEntity} name of the field from which to search.
+     * @param value   The value of the field from which to search.
      * @return The {@link KapuaEntity} found, or {@code null} if not found.
+     * @throws NonUniqueResultException When more than one result is returned
      * @since 1.0.0
      */
-    public static <E extends KapuaEntity> E findByField(@NotNull EntityManager em, @NotNull Class<E> clazz, @NotNull String name, @NotNull String value) {
+    @Nullable
+    public static <E extends KapuaEntity> E findByField(@NotNull EntityManager em,
```

Comment:
I would have kept 

- `findByField(em, clazz, name, value) `
- `findByField(em, clazz, scopeId name, value)`

in order to avoid to pass `null` to a methods which usage could be confusing. 
Also they actually have a slight difference which is that the one without `scopeId` needs checkaccess perfomed after invokation while with `scopeId` check access should have already happened